### PR TITLE
[SPARK-48482][PYTHON] dropDuplicates and dropDuplicatesWIthinWatermark should accept variable length args

### PR DESCRIPTION
**Pull request by WweiL**
### What changes were proposed in this pull request?

In scala, `dropDuplicates` and `dropDuplicatesWIthinWatermark` accepts varargs, i.e. `df.dropDuplicates("id", "value")`. However this is not supported in Python, users have to wrap them with list. This PR fixes it.

### Why are the changes needed?

Better API, integrated scala and python experience

### Does this PR introduce _any_ user-facing change?

Yes, now users can use var args as parameters to `dropDuplicates` and `dropDuplicatesWithinWatermark`

### How was this patch tested?

Added & modified existing unit tests

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/classic/dataframe.py [hunk 1]
```diff
@@ -1220,28 +1220,20 @@ def intersectAll(self, other: ParentDataFrame) -> ParentDataFrame:
...
 
-    def dropDuplicates(self, subset: Optional[List[str]] = None) -> ParentDataFrame:
-        if subset is not None and (not isinstance(subset, Iterable) or isinstance(subset, str)):
-            raise PySparkTypeError(
-                error_class="NOT_LIST_OR_TUPLE",
-                message_parameters={"arg_name": "subset", "arg_type": type(subset).__name__},
-            )
-
-        if subset is None:
+    def dropDuplicates(self, *subset: Union[str, List[str]]) -> ParentDataFrame:
```
Comment (HyukjinKwon) [inline python/pyspark/sql/classic/dataframe.py:1223]: Let's also fix `docs/structured-streaming-programming-guide.md` together :-)

**Comment by WweiL** [comment]
will take a further look for lint error

**Comment by WweiL** [comment]
I think this is good to be merged : ) @HyukjinKwon

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**